src/shared/utils/traverseAllChildren.js

```diff
@@ -111,54 +111,16 @@ function traverseAllChildrenImpl(
     if (iteratorFn) {
       var iterator = iteratorFn.call(children);
       var step;
-      if (iteratorFn !== children.entries) {
```

Comment:
Maybe we should still warn in the case `iteratorFn !== children.entries` in case we want to have this back? People shouldn't use maps where the key is implicitly another child. Could create strange bugs.